composite.go

```diff
@@ -96,42 +97,39 @@ func createCompositeRequestForCollection(method string, url string, allOrNone bo
 	}, nil
 }
 
-func processCompositeResponse(resp http.Response) error {
+func processCompositeResponse(resp http.Response, allOrNone bool) (*SalesforceResults, error) {
 	compositeResults := compositeRequestResult{}
-	var errorResponse error
+	results := SalesforceResults{}
 
 	responseData, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	jsonError := json.Unmarshal(responseData, &compositeResults)
 	if jsonError != nil {
-		return jsonError
+		return nil, jsonError
 	}
 
 	for _, subResult := range compositeResults.CompositeResponse {
-		for _, sfError := range subResult.Body {
-			if !sfError.Success {
-				if len(sfError.Errors) > 0 {
-					for _, errorMessage := range sfError.Errors {
-						newError := errorMessage.StatusCode + ": " + errorMessage.Message + " " + sfError.Id
-						errorResponse = errors.Join(errorResponse, errors.New(newError))
-					}
-				} else {
-					newError := "an unknown error occurred: " + strconv.Itoa(subResult.HttpStatusCode)
-					errorResponse = errors.Join(errorResponse, errors.New(newError))
-				}
+		for _, result := range subResult.Body {
+			if !result.Success {
+				results.HasErrors = true
 			}
 		}
+		results.Results = append(results.Results, subResult.Body...)
+	}
+
+	if results.HasErrors && allOrNone {
+		fmt.Println("Records rolled back because not all records were valid and the request was using AllOrNone header")
```

Comment:
Wording suggestion

```suggestion
		fmt.Println("Records rolled back because some records were invalid and the request was using AllOrNone header")
```

Reply:
Keeping as is only because I copied and pasted from Salesforce's own wording. So just makes it consistent.